src/renderer/dx/CustomTextRenderer.cpp

```diff
@@ -254,13 +255,17 @@ void CustomTextRenderer::_FillRectangle(void* clientDrawingContext,
         rect.right += glyphRun->glyphAdvances[i];
     }
 
-    d2dContext4->FillRectangle(rect, drawingContext->backgroundBrush);
+    d2dContext->FillRectangle(rect, drawingContext->backgroundBrush);
 
     // Now go onto drawing the text.
 
     // First check if we want a color font and try to extract color emoji first.
-    if (WI_IsFlagSet(drawingContext->options, D2D1_DRAW_TEXT_OPTIONS_ENABLE_COLOR_FONT))
+    // Color emoji are only available on Windows 10+
+    if (WI_IsFlagSet(drawingContext->options, D2D1_DRAW_TEXT_OPTIONS_ENABLE_COLOR_FONT) && IsWindows10OrGreater())
     {
+        ::Microsoft::WRL::ComPtr<ID2D1DeviceContext4> d2dContext4;
+        RETURN_IF_FAILED(drawingContext->renderTarget->QueryInterface(d2dContext4.GetAddressOf()));

```

Comment:
If we have `renderTarget` somewhere as a `ComPtr` i'd prefer we use `renderTarget.As(&d2dContext4)`; is that possible?
If we don't, that's probably fine, but we should clean this up later.

Reply:
In general, we shouldn't use `QueryInterface` directly :smile:

Reply:
I'm not sure, I'll test it. I was following the pattern that was used previously.